dataset_conversion/utils.py

```diff
@@ -0,0 +1,781 @@
+import os
+import nibabel as nib
+import numpy as np
+import logging
+from copy import deepcopy
+import subprocess
+
+logger = logging.getLogger(__name__)
+
+
+def binarize_label(subject_path, label_path):
+    label_npy = nib.load(label_path).get_fdata()
+    threshold = 0.5
+    label_npy = np.where(label_npy > threshold, 1, 0)
+    ref = nib.load(subject_path)
+    label_bin = nib.Nifti1Image(label_npy, ref.affine, ref.header)
+    # overwrite the original label file with the binarized version
+    nib.save(label_bin, label_path)
+
+
+def create_region_based_label(lesion_label_file, seg_label_file, image_file, sub_ses_name, thr=0.5):
+    """
+    Creates region-based labels for nnUNet training. The regions are: 
+    0: background
+    1: spinal cord seg
+    2: lesion seg
+    """
+    # load the labels
+    lesion_label_npy = nib.load(lesion_label_file).get_fdata()
+    seg_label_npy = nib.load(seg_label_file).get_fdata()
+
+    # binarize the labels
+    lesion_label_npy = np.where(lesion_label_npy > thr, 1, 0)
+    seg_label_npy = np.where(seg_label_npy > thr, 1, 0)
+
+    # check if the shapes of the labels match
+    assert lesion_label_npy.shape == seg_label_npy.shape, \
+          f'Shape mismatch between lesion label and segmentation label for subject {sub_ses_name}. Check the labels.'
+
+    # create a new label array with the same shape as the original labels
+    label_npy = np.zeros(lesion_label_npy.shape, dtype=np.int16)
+    # spinal cord
+    label_npy[seg_label_npy == 1] = 1
+    # lesion seg
+    label_npy[lesion_label_npy == 1] = 2
```

Comment:
@naga-karthik I was thinking about using `sct_maths -add` to make sure that the lesion is part of the SC seg (as we did this for SCIseg). But now I am reading these two lines. Basically, line 45 sets the value `2` to the lesion regardless of whether the lesion is part of the SC or not. Meaning that `sct_maths -add` is unnecessary?

Reply:
Yeah, you're right! Because this was already happening during nnunet conversion, I thought that you were doing `sct_maths -add` just as a sanity check to ensure lesion is inside the SC!

Reply:
Okay, great, thanks for confirmation!